helper/schema/resource.go

```diff
@@ -69,6 +96,13 @@ type DeleteFunc func(*ResourceData, interface{}) error
 // See Resource documentation.
 type ExistsFunc func(*ResourceData, interface{}) (bool, error)
 
+// See Resource documentation for more details.
+type StateMigrationContext struct {
+	StateSchemaVersion int
+	State              *terraform.InstanceState
+}
+type StateMigrateFunc func(*StateMigrationContext, interface{}) error
```

Comment:
I can't think of any future reason why we'd want to augment the context. I think we should just use the `(int, state, interface{})` directly. I like that API better.